### PR TITLE
Fix #17 issue, adding ObservableApplicationVersion class, changing Ap…

### LINKED ISSUE #17: ApplicationFacet class
- It is possible to use observable:applicationIdentifier instead of uco-core:name, even though the new property does not have an explicit semantic. Alternatively also the core:name can be used but outside the uco-core:hasFacet array.
- Example:
``` json
{
            "@id": "kb:0d191a90-7f5f-4df7-9cc3-3b5024e66983",
            "@type": "uco-observable:ObservableObject",
            "uco-core:name": "Native application",
            "uco-core:hasFacet": [
                {
                    "@id": "kb:f35ce51a-cb45-49ca-83d1-546cedd73b21",
                    "@type": "uco-observable:ApplicationFacet",
                    "uco-observable:applicationIdentifier": "Native"
                }
            ]
    }
```

### PR DESCRIPTION
**Pull request by fabrizio-turchi**
@ajnelson-nist I changed the Bundle class in core module, by defining a new "property" self["uco-core:UcoInherentCharacterizationThing"] = [] and adding the append_to_uco_inherent_characterization_thing(self, *args) method. Therefore instead of defining the super classes (UcoThing and UcoInherentCharacterizationThing) I thought that the changes I made could be a simpler solution. If you think it would be better to adopt another solution let me know, I will gladly do so!

### CODE REVIEW

**Comment by fabrizio-turchi** [comment]
@ajnelson-nist A slightly better solution would be to avoid the statement:

```python
self["uco-core:UcoInherentCharacterizationThing"] = []
```

and change the append_to_uco_inherent_characterization_thing method like this:

```python
if not self.get("uco-core:UcoInherentCharacterizationThing"):
            self["uco-core:UcoInherentCharacterizationThing"] = []
        
        self._append_observable_objects(
            "uco-core:UcoInherentCharacterizationThing", *args
        )
```

File: case.jsonld
Comment (ajnelson-nist) [inline case.jsonld:24]: `uco-core:UcoInherentCharacterizationThing` is a class, not a property.

I'll add a revision patch.

Comment (ajnelson-nist) [reply case.jsonld]: Revised.  This required some moderate scope-creep of the PR, so I probably shouldn't be the last review before merge.

File: case.jsonld
Comment (ajnelson-nist) [inline case.jsonld:1601]: This should be `PATH`.

File: case.jsonld
Comment (ajnelson-nist) [inline case.jsonld:1600]: I didn't see this class in `types.py`.  Since the object's being generated now, it should be added to the API.

File: case.jsonld
Comment (ajnelson-nist) [inline case.jsonld:1596]: I didn't see this class in `types.py`. Since the object's being generated now, it should be added to the API.

File: case.jsonld
Comment (ajnelson-nist) [reply case.jsonld:1601]: Oh, maybe I'm mistaken?  I forgot what OS was being characterized.

File: case_mapping/uco/observable.py
Comment (ajnelson-nist) [inline case_mapping/uco/observable.py:428]: Review note: This class definition needed to move below `ObservableObject`'s definition for a type-signature reference.

File: case.jsonld
Comment (ajnelson-nist) [reply case.jsonld:1596]: Implemented in 6cb5844.

File: case.jsonld
Comment (ajnelson-nist) [reply case.jsonld:1600]: Implemented in 6cb5844.

File: case.jsonld
Comment (ajnelson-nist) [reply case.jsonld:1601]: I won't block review on this, but it's worth double-checking this with instance data.

**Review by ajnelson-nist — APPROVED** [review]
In reviewing the newly-generated JSON-LD data, I saw reason to make significant revision to the base class hierarchy and port some prior implementation into generic classes in `types.py`.

The extra implementation suggests some other changes that should be just outside the scope of this PR:

* There are some other opportunities to extend the `types.Dictionary` class to cover prior generated example data related to EXIF.
* `types.Hash` should also be implemented so we can decide some code-design matters around when to fully inline nodes vs. make a stub reference.

In light of the scope-creep, I think someone else should review and merge this PR now.

**Review by kchason — APPROVED** [review]

**Pull request merged**
